packaging/os/yumrepo.py

```diff
@@ -0,0 +1,501 @@
+    params = None
+    section = None
+    filename = None
+    repofile = ConfigParser.RawConfigParser()
+
+    def __init__(self, module, repo_dir):
+        # To be albe to use fail_json
+        self.module = module
+        # Shortcut for the params
+        self.params = self.module.params
+        # Section is always the repoid
+        self.section = self.params['repoid']
+
+        # Check if repo directory exists
+        if not os.path.isdir(repo_dir):
+            self.module.fail_json(
+                msg='Repo directory "%s" does not exist.' % repo_dir)
+
+        # Get the given or the default repo file name
+        repo_file = self.params['repoid']
+        if self.params['file'] is not None:
+            repo_file = self.params['file']
+
+        # Set filename
+        self.filename = "%s/%s.repo" % (repo_dir, repo_file)
+
+        # Read the repo file if it exists
+        if os.path.isfile(self.filename):
+            self.repofile.read(self.filename)
+
+    def add(self):
+        # Remove already existing repo and create a new one
+        if self.repofile.has_section(self.section):
+            self.repofile.remove_section(self.section)
+
+        # Add section
+        self.repofile.add_section(self.section)
+
+        # Baseurl/mirrorlist is not required because for removal we need only
+        # the repo name. This is why we check if the baseurl/mirrorlist is
+        # defined.
+        if (self.params['baseurl'], self.params['mirrorlist']) == (None, None):
+            self.module.fail_json(
+                msg='Paramater "baseurl" or "mirrorlist" is required for '
+                'adding a new repo.')
+
+        # Set options
+        for key in sorted(self.params.keys()):
+            value = self.params[key]
+
+            # Convert boolean value to integer
+            if type(value).__name__ == 'bool':
+                value = int(value)
+
+            # Set the value only if it was defined (default is None)
+            if value is not None and key not in ['file', 'repoid', 'state']:
+                self.repofile.set(self.section, key, value)
+
+    def save(self):
+        if len(self.repofile.sections()):
+            # Write data into the file
+            fd = open(self.filename, 'wb')
+            self.repofile.write(fd)
+            fd.close()
+        else:
+            # Remove the file if there are not repos
+            os.remove(self.filename)
+
+    def remove(self):
+        # Remove section if exists
+        if self.repofile.has_section(self.section):
+            self.repofile.remove_section(self.section)
+
+    def dump(self):
+        repo_string = ""
+
+        # Compose the repo file
+        for section in sorted(self.repofile.sections()):
+            repo_string += "[%s]\n" % section
+
+            for key, value in self.repofile.items(section):
+                repo_string += "%s = %s\n" % (key, value)
+
+            repo_string += "\n"
+
+        return repo_string
```

Comment:
On second thought I'm not sure that would sort the sections the right way, especially if you remove and add back the same section.  Hm.

On the other hand, you're not sorting the keys.